.ycm_extra_conf.py

```diff
@@ -1,67 +1,250 @@
+# to the requirements of the PEP 8 style guide and 'systemd' project,
+# respectively.
+#
+# The original license is preserved as it is.
+#
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# For more information, please refer to <http://unlicense.org/>
+
+"""
+YouCompleteMe configuration file tailored to support the 'meson' build system
+used by the 'systemd' project.
+"""
+
+import glob
 import os
-import subprocess
+import ycm_core
 
-def GetFlagsFromMakefile(varname):
-  return subprocess.check_output([
-      "make", "-s", "print-%s" % varname]).decode().split()
 
-
-def Flatten(lists):
-  return list(itertools.chain.from_iterable(lists))
+SOURCE_EXTENSIONS = (".C", ".cpp", ".cxx", ".cc", ".c", ".m", ".mm")
+HEADER_EXTENSIONS = (".H", ".h", ".hxx", ".hpp", ".hh")
 
 
 def DirectoryOfThisScript():
-  return os.path.dirname(os.path.abspath(__file__))
+    """
+    Return the absolute path of the parent directory containing this
+    script.
+    """
+    return os.path.dirname(os.path.abspath(__file__))
+
+
+def GuessBuildDirectory():
+    """
+    Guess the build directory using the following heuristics:
+
+    1. Returns the current directory of this script plus 'build'
+    subdirectory in absolute path if this subdirectory exists.
+
+    2. Otherwise, probes whether there exists any directory
+    containing '.ninja_log' file two levels above the current directory;
+    returns this single directory only if there is one candidate.
+    """
+    result = os.path.join(DirectoryOfThisScript(), "build")
+
+    if os.path.exists(result):
+        return result
+
+    result = glob.glob(os.path.join(DirectoryOfThisScript(),
+                                    "..", "..", "*", ".ninja_log"))
+
+    if not result:
+        return ""
+
+    if 1 != len(result):
```

Comment:
Yikes. That meme is horrible even in C where it superficially made sense years ago (an assignment is a valid expression), but there's no reason even to use it in Python.